packages/lit/src/components/wizard-token-navigator/index.ts

```diff
@@ -0,0 +1,96 @@
+import { LitElement, html, TemplateResult } from 'lit';
+import { t } from '../../i18n/messages';
+import { ShadowDOMNavigator } from '../../lib/ShadowDOMNavigator';
+
+const FOCUS_DELAY_MS = 300;
+const TOKEN_FIELD_SELECTOR = 'wizard-token-field';
+
+/**
+ * Abstract base class for components that need token navigation functionality
+ * Provides token field navigation, focus management, and token link rendering
+ */
+export abstract class WizardTokenNavigator extends LitElement {
+  readonly #shadowDOMNavigator = new ShadowDOMNavigator();
+
+  /**
+   * 1. Finds the field (crossing shadow DOM boundaries)
+   * 2. Expands any collapsed parent <details> elements
+   * 3. Applies highlight animation
+   * 4. Scrolls field into view
+   * 5. Focuses the input element
+   * @param tokenPath - Path to the token (e.g., "basis.color.primary")
+   */
+  protected focusToken(tokenPath: string): void {
+    const selector = `${TOKEN_FIELD_SELECTOR}[path="${tokenPath}"]`;
+    const field = this.#shadowDOMNavigator.findElement(document, selector);
+
+    if (!field) {
+      console.warn(`[ValidationIssues] Token field not found: ${tokenPath}`);
+      return;
+    }
+
+    this.#shadowDOMNavigator.expandParentDetails(field);
+    this.applyHighlightAnimation(field);
+    this.scrollToField(field);
+    this.focusFieldInput(field, tokenPath);
+  }
+
+  /**
+   * Applies a highlight animation to the field
+   */
+  protected applyHighlightAnimation(field: HTMLElement): void {
+    field.classList.add('theme-validation-highlight');
+
+    const handleAnimationEnd = () => {
+      field.classList.remove('theme-validation-highlight');
+      field.removeEventListener('animationend', handleAnimationEnd);
+    };
+
+    field.addEventListener('animationend', handleAnimationEnd);
+  }
+
+  /**
+   * Scrolls the field into view
+   */
+  protected scrollToField(field: HTMLElement): void {
+    const prefersReducedMotion = globalThis.matchMedia('(prefers-reduced-motion: reduce)').matches;
+    field.scrollIntoView({
+      behavior: prefersReducedMotion ? 'auto' : 'smooth',
+      block: 'center',
+    });
+  }
+
+  /**
+   * Uses setTimeout to allow scroll animation to complete first
+   */
+  protected focusFieldInput(field: HTMLElement, tokenPath: string): void {
+    const inputId = `input-${tokenPath}`;
+    const input = field.shadowRoot?.getElementById(`${inputId}`)?.shadowRoot?.querySelector('input');
```

Comment:
Dit lijkt me n beetje onveilige aanname. Ja de meeste van die elementen hebben dit, maar het is niet gezegd dat ze die allemaal blijven hebben. Twee manieren om te op te lossen:
1. centrale functie om die IDs op te bouwen, zodat je het consistent doet. Importeren overal waar je t nodig hebt
2. Selector op basis van het path attribute, dit zijn namelijk per definitie dezelfde waardes.